src/java.base/share/classes/java/util/stream/SliceOps.java

```diff
@@ -110,9 +105,17 @@ private static long calcSliceFence(long skip, long limit) {
                                         long skip, long limit) {
         if (skip < 0)
             throw new IllegalArgumentException("Skip must be non-negative: " + skip);
+        long adjustedLimit = limit >= 0 ? limit : Long.MAX_VALUE;
 
         return new ReferencePipeline.StatefulOp<T, T>(upstream, StreamShape.REFERENCE,
                                                       flags(limit)) {
+            @Override
+            long adjustSize(long size) {
+                // For parallel streams, spliterator size is already adjusted
+                return isParallel() ? super.adjustSize(size)
+                    : calcSize(super.adjustSize(size), skip, adjustedLimit);
+            }
+
```

Comment:
```suggestion
            @Override
            long exactOutputSize(long sourceSize) {
                // For parallel streams, the exact output size is that reported by the slice op's spliterator,
                // since a slice op is a stateful op whose spliterator is the source spliterator
                // (see AbstractPipeline.sourceSpliterator).
                // Otherwise, for sequential streams the extract output size that of the prior pipeline stage reduced
                // by the skip and limit
                return isParallel() ? sourceSize
                    : calcSize(super.exactOutputSize(sourceSize), skip, normalizedLimit);
            }
```